### ISSUE TITLE
pundit_role cannot be set on array of arguments.

### ISSUE THREAD

**Issue by thijsnado** [rmosolgo/graphql-ruby#3146]
**Describe the bug**

Run into namespacing error when trying to specify pundit role on array of AR records on argument for mutation. Given the following ruby:

```ruby
# frozen_string_literal: true
class Mutations::AMutationOnUser < Mutations::BaseMutation
  argument :user_ids, [ID], required: true, loads: Types::UserType, pundit_role: :update

  # rest of code
end
```

Instead of making a policy for each user it loads, it tries to get the policy for the array. I think this causes https://github.com/varvet/pundit#policy-namespacing to get triggered giving an error like

```
 Pundit::NotDefinedError: unable to find policy `User::UserPolicy`
```

**Versions**

`graphql` version: 1.11.4
`rails` (or other framework): 5.2.4.3
`pundit`: 2.1.0

**Comment by rmosolgo** [comment]
Thanks for the detailed report, I'll take a look!

**Comment by thornomad** [comment]
I believe I am having the same issue and wanted to add slightly different context if it helps.

```ruby
module Mutations
  class MyMutation < Mutations::BaseMutation
    type Boolean

    argument :ids, [ID], required: true, as: :schedules, loads: Types::ScheduleType, pundit_role: :update

    def resolve(schedules:)
      true
    end

    def self.pundit_policy_class
      ::SchedulePolicy
    end
  end
end
```

I am not receiving a `Pundit::NotDefinedError` (as above); the policy is found.  The issue is that when the policy rejects the permissions the error being raised is not handled in the same way as a one-off policy.  Usually, with a one-off record it will call `MyApiSchema.unauthorized_object`.  However, in this case it is raising a `GraphQL::UnauthorizedError` which ends up having a slightly different result with the message of "An instance of Array failed ids's authorization check"

Here is a backtrace if that helps:

```
"/Users/damon/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/graphql-pro-1.15.4/lib/graphql/pro/pundit_integration.rb:208:in `unauthorized_by_pundit'
/ruby/gems/2.7.0/gems/graphql-pro-1.15.4/lib/graphql/pro/pundit_integration.rb:198:in `block in authorized?'
/ruby/gems/2.7.0/gems/graphql-pro-1.15.4/lib/graphql/pro/pundit_integration.rb:175:in `each'
/ruby/gems/2.7.0/gems/graphql-pro-1.15.4/lib/graphql/pro/pundit_integration.rb:175:in `all?'
/ruby/gems/2.7.0/gems/graphql-pro-1.15.4/lib/graphql/pro/pundit_integration.rb:175:in `authorized?'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/schema/resolver.rb:85:in `block (2 levels) in resolve_with_support'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/schema.rb:112:in `after_lazy'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/schema.rb:109:in `block in after_lazy'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/lazy.rb:41:in `value'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/schema.rb:123:in `public_send'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/schema.rb:123:in `sync_lazy'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/schema.rb:107:in `block in after_lazy'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/lazy.rb:41:in `value'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/schema.rb:123:in `public_send'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/schema.rb:123:in `sync_lazy'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter/runtime.rb:439:in `block (3 levels) in after_lazy'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/tracing.rb:66:in `trace'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter/runtime.rb:438:in `block (2 levels) in after_lazy'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/query.rb:353:in `block in with_error_handling'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/errors.rb:41:in `with_error_handling'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/query.rb:352:in `with_error_handling'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter/runtime.rb:436:in `block in after_lazy'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/lazy.rb:41:in `value'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter/resolve.rb:37:in `resolve'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter/resolve.rb:11:in `resolve_all'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter/runtime.rb:245:in `block (2 levels) in evaluate_selections'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter/runtime.rb:458:in `after_lazy'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter/runtime.rb:177:in `block in evaluate_selections'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter/runtime.rb:124:in `each'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter/runtime.rb:124:in `evaluate_selections'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter/runtime.rb:60:in `run_eager'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter.rb:73:in `block in evaluate'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/tracing.rb:66:in `trace'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter.rb:72:in `evaluate'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/interpreter.rb:45:in `begin_query'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/multiplex.rb:112:in `begin_query'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/multiplex.rb:83:in `block in run_as_multiplex'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/multiplex.rb:82:in `map'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/multiplex.rb:82:in `run_as_multiplex'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/multiplex.rb:61:in `block (2 levels) in run_queries'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/multiplex.rb:195:in `block in instrument_and_analyze'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/instrumentation.rb:29:in `block (2 levels) in apply_instrumenters'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/instrumentation.rb:46:in `block (2 levels) in each_query_call_hooks'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/instrumentation.rb:41:in `each_query_call_hooks'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/instrumentation.rb:45:in `block in each_query_call_hooks'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/instrumentation.rb:72:in `call_hooks'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/instrumentation.rb:44:in `each_query_call_hooks'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/instrumentation.rb:27:in `block in apply_instrumenters'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/instrumentation.rb:72:in `call_hooks'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/instrumentation.rb:26:in `apply_instrumenters'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/multiplex.rb:174:in `instrument_and_analyze'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/multiplex.rb:60:in `block in run_queries'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/tracing.rb:66:in `trace'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/multiplex.rb:58:in `run_queries'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/execution/multiplex.rb:48:in `run_all'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/schema.rb:1642:in `multiplex'
/ruby/gems/2.7.0/gems/graphql-1.11.4/lib/graphql/schema.rb:1613:in `execute'
p/controllers/graphql_controller.rb:32:in `execute'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_controller/metal/basic_implicit_render.rb:6:in `send_action'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/abstract_controller/base.rb:195:in `process_action'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_controller/metal/rendering.rb:30:in `process_action'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/abstract_controller/callbacks.rb:42:in `block in process_action'
/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:135:in `run_callbacks'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/abstract_controller/callbacks.rb:41:in `process_action'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_controller/metal/rescue.rb:22:in `process_action'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_controller/metal/instrumentation.rb:33:in `block in process_action'
/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/notifications.rb:180:in `block in instrument'
/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/notifications.rb:180:in `instrument'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_controller/metal/instrumentation.rb:32:in `process_action'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_controller/metal/params_wrapper.rb:245:in `process_action'
/ruby/gems/2.7.0/gems/activerecord-6.0.3.3/lib/active_record/railties/controller_runtime.rb:27:in `process_action'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/abstract_controller/base.rb:136:in `process'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_controller/metal.rb:190:in `dispatch'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_controller/metal.rb:254:in `dispatch'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/routing/route_set.rb:50:in `dispatch'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/routing/route_set.rb:33:in `serve'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/journey/router.rb:49:in `block in serve'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/journey/router.rb:32:in `each'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/journey/router.rb:32:in `serve'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/routing/route_set.rb:834:in `call'
/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/etag.rb:27:in `call'
/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/conditional_get.rb:40:in `call'
/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/head.rb:12:in `call'
/ruby/gems/2.7.0/gems/activerecord-6.0.3.3/lib/active_record/migration.rb:567:in `call'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/callbacks.rb:101:in `run_callbacks'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/middleware/callbacks.rb:26:in `call'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/middleware/executor.rb:14:in `call'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/middleware/actionable_exceptions.rb:17:in `call'
/ruby/gems/2.7.0/gems/appsignal-2.11.0/lib/appsignal/rack/rails_instrumentation.rb:19:in `call'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/middleware/debug_exceptions.rb:32:in `call'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
/ruby/gems/2.7.0/gems/railties-6.0.3.3/lib/rails/rack/logger.rb:37:in `call_app'
/ruby/gems/2.7.0/gems/railties-6.0.3.3/lib/rails/rack/logger.rb:26:in `block in call'
/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/tagged_logging.rb:80:in `block in tagged'
/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/tagged_logging.rb:28:in `tagged'
/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/tagged_logging.rb:80:in `tagged'
/ruby/gems/2.7.0/gems/railties-6.0.3.3/lib/rails/rack/logger.rb:26:in `call'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
/ruby/gems/2.7.0/gems/request_store-1.5.0/lib/request_store/middleware.rb:19:in `call'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/middleware/request_id.rb:27:in `call'
/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/runtime.rb:22:in `call'
/ruby/gems/2.7.0/gems/activesupport-6.0.3.3/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/middleware/executor.rb:14:in `call'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/middleware/static.rb:126:in `call'
/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/sendfile.rb:110:in `call'
/ruby/gems/2.7.0/gems/actionpack-6.0.3.3/lib/action_dispatch/middleware/host_authorization.rb:82:in `call'
/ruby/gems/2.7.0/gems/hirefire-resource-0.4.2/lib/hirefire/middleware.rb:41:in `call'
/ruby/gems/2.7.0/gems/rack-cors-1.0.3/lib/rack/cors.rb:95:in `call'
/ruby/gems/2.7.0/gems/railties-6.0.3.3/lib/rails/engine.rb:527:in `call'
/ruby/gems/2.7.0/gems/puma-4.3.6/lib/puma/configuration.rb:228:in `call'
/ruby/gems/2.7.0/gems/puma-4.3.6/lib/puma/server.rb:713:in `handle_request'
/ruby/gems/2.7.0/gems/puma-4.3.6/lib/puma/server.rb:472:in `process_client'
/ruby/gems/2.7.0/gems/puma-4.3.6/lib/puma/server.rb:328:in `block in run'
/ruby/gems/2.7.0/gems/puma-4.3.6/lib/puma/thread_pool.rb:134:in `block in spawn_thread'
```
If this is another issue entirely my apologies and I can open a new one.

**Comment by rmosolgo** [comment]
Hey @thijsnado, thanks again for the detailed report. It turns out it only happened when an explicit `pundit_role:` is configured on the argument, which is why I hadn't noticed it before. I just shipped graphql-pro 1.15.7 which includes a fix for this issue, please let me know if you run into anything else! 

@thornomad, I'm not exactly sure if your issue was resolved or not. If you're still running into trouble, please open a new issue and let me know!
